src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/System.Text.Json.SourceGeneration.Tests.targets

```diff
@@ -112,6 +112,10 @@
     <Compile Include="$(CoreLibSharedDir)System\Diagnostics\CodeAnalysis\DynamicallyAccessedMemberTypes.cs" />
   </ItemGroup>
 
+  <ItemGroup>
+    <ProjectReference Include="$(LibrariesProjectRoot)Microsoft.Extensions.Primitives\src\Microsoft.Extensions.Primitives.csproj" />
```

Comment:
This doesn't feel right to me. I know these are tests, but having something in `System` depend on things in `Microsoft.Extensions` feels backwards.

Is there something about `StringValues` that we can isolate into a unit test type? Or do we actually have to use the `StringValues` type in order to test the scenario?

Reply:
StringValues is what customer reported so it was added here as a test to make sure that specific scenario works. I think that should be fine for tests, we already depend on other things like JSON.NET